File: c#/ordQuick/ordQuick.cs
using System.Globalization;
using Microsoft.VisualBasic;

class Ord_Quick
{
    public static (int[],int) Partition(int[] a, int l, int h)
    {
        int pvt = a[h];
        int j = l - 1;
        for (int k = l; k < h; k++)
        {
            if (a[k] < pvt)
            {
                j++;
                a = Swap(a, j, k);
            }
        }
        a = Swap(a, j + 1, h);

        return (a, j+1);

    }

    public static int[] Swap(int[] a, int j, int k)
    {
        int adicional = 0;
        adicional = a[j];
        a[j] = a[k];
        a[k] = adicional;
        return a;
    }

    public static int[] OrdQuick(int[] a, int l, int h)
    {
        if (l < h)
        {
            var valores = Partition(a, l, h);
            a = OrdQuick(valores.Item1, l, valores.Item2 - 1);
            a = OrdQuick(valores.Item1, valores.Item2 + 1, h);
        }
        return a;
    }

    public static void Mostrar(int[] a)
    {
        for (int i = 0; i < a.Length; i++)
        {
            Console.Write(a[i] + " ");
        }
        Console.WriteLine();
    }
    public static void Main(String[] args)
    {
        int[] a = [10, 7, 8, 9, 1, 5];
        Mostrar(a);
        OrdQuick(a, 0, a.Length - 1);
        Mostrar(a);
    }
}

File: jv/ordQuick.java
public class ordQuick{
    public static void show(int[] a){
        for (Object x : a) {
            System.out.print(x + " ");
        }
        System.out.println("");
    }

    public static void swap(int[] a, int k, int j){
        int adicional = a[k];
        a[k] = a[j];
        a[j] = adicional;
    }

    public static int partition(int[] a, int l, int h){
        int pvt = a[h];
        int j = l-1;

        for(int k=l;k<h;k++){
            if(a[k]<pvt){
                j++;
                swap(a,k,j);
            }
        }
        swap(a, j+1, h);
        return j+1;
    }

    public static int[] qckSort(int[] a, int l, int h){
        
        if(l<h){
            int piv = partition(a, l, h);

            qckSort(a, l, piv-1);
            qckSort(a, piv+1, h);
        }
        return a;
    }

    public static void main(String[] args) {
        int[] a = {10, 7, 8, 9, 1, 5};
        int l = 0, h= a.length-1;
        show(a);
        a = qckSort(a, l, h);
        show(a);
    }

}